src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -621,11 +629,12 @@ export default class QueryStringInputUI extends Component<Props, State> {
         selectionStart: null,
         selectionEnd: null,
       });
-      if (document.activeElement !== null && document.activeElement.id === this.textareaId) {
-        this.handleAutoHeight();
-      } else {
-        this.handleRemoveHeight();
-      }
+    }
+
+    if (document.activeElement !== null && document.activeElement.id === this.textareaId) {
```

Comment:
I agree that is easier to read, but logically this is a bit different. `undefined` === `undefined ` case is different. 
I'd leave it as is, because need to review this logical case first to make this change